packages/curve-ui-kit/src/lib/model/query/factory.ts

```diff
@@ -55,6 +55,15 @@ export function queryFactory<
     setQueryData: (params, data) => queryClient.setQueryData<TData>(queryKey(params), data),
     prefetchQuery: (params, staleTime = 0) => queryClient.prefetchQuery({ ...getQueryOptions(params), staleTime }),
     fetchQuery: (params, options) => queryClient.fetchQuery({ ...getQueryOptions(params), ...options }),
+    /**
+     * Function that is like fetchQuery, but first invalidates the query to ensure fresh data is fetched.
+     * Primary use case is for Zustand stores where want to to both use queries and ensure freshness.
+     * I suspect this will be the only case, and once Zustand refactoring to Tanstack is complete, we may delete this.
+     */
+    refetchQuery: async (params) => {
+      await queryClient.invalidateQueries({ queryKey: queryKey(params) })
+      return queryClient.fetchQuery({ ...getQueryOptions(params), ...options })
+    },
```

Comment:
Maybe this works too?
```suggestion
    refetchQuery: (params) => queryClient.fetchQuery({ ...getQueryOptions(params), ...options, staleTime: 0 }),
```
<img width="685" height="172" alt="image" src="https://github.com/user-attachments/assets/0edbd2dc-886a-4a0e-83a7-fe7fab827bde" />